sbin/wifibox

```diff
@@ -703,9 +709,39 @@ vm_manager() {
 
     log debug "Nice priority: ${_nice_priority}"
     log debug "Arguments: ${_bhyve_args}"
-    ${NICE} -n ${_nice_priority} \
-	    ${BHYVE} ${_bhyve_args} 2>&1 | capture_output debug bhyve
+
+    _kill_child() {
+    	local _job_pids
+    	_job_pids="$(jobs -p)"
+
+    	log debug "SIGTERM received for PID $PPID, killing remaining jobs"
+
+    	if [ -n "${_job_pids}" ]; then
+    	    for job_pid in ${_job_pids}; do
+    		log debug "job_pid=${job_pid}"
+    		${KILL} -SIGTERM ${job_pid} | capture_output debug kill
+    	    done
+    	fi
+    }
+
+    trap _kill_child SIGTERM
+
+    { ${NICE} -n ${_nice_priority} \
+	    ${BHYVE} ${_bhyve_args} 2>&1 | capture_output debug bhyve; } &
+    _nice_pid=$!
+
+    log debug "Waiting for VM to stop"
+
+    wait "$_nice_pid"
     _bhyve_exit_code="$?"
+
+    trap - SIGTERM
+
+    if [ "${_bhyve_exit_code}" -eq "143" ]; then
+	wait "$_nice_pid"
+	_byve_exit_code="$?"
+    fi
+
```

Comment:
The value of `_bhyve_exit_code` should be logged (with `log debug`) for tracing purposes.